gspread/spreadsheet.py

```diff
@@ -287,11 +287,11 @@ def get_worksheet_by_id(self, id):
 
         try:
             item = finditem(
-                lambda x: x["properties"]["sheetId"] == id,
+                lambda x: x["properties"]["sheetId"] == int(id),
```

Comment:
as discussed, I think this should be `str(id)`

Reply:
The `sheet_data` object in this function stores `sheet_data["properties"]["sheetId"]` as an int, passing a string into the lambda fails every time.  Seems like the upstream `fetch_sheet_metadata()` call returns the G Sheet API response unmodified and [according to the Google Documentation](https://developers.google.com/sheets/api/reference/rest/v4/spreadsheets/sheets#SheetProperties), the `sheetId` value will always be an integer so I think string -> int is the way to go.

Reply:
you're right, I wasn't sure of Google's stance. in that case what you had is perfect! and perhaps we should change delete to send the data as an int.

good investigating!

Reply:
Sound's good, the [API declares](https://developers.google.com/sheets/api/reference/rest/v4/spreadsheets/request#DeleteSheetRequest) that a `DeleteSheetRequest` should also provide `sheetId` as an int so perhaps it's just been luck it's been working all this time.

Reply:
I think so, yes. Let's switch to int